zp-relayer/workers/sentTxWorker.ts

```diff
@@ -140,18 +143,26 @@ export async function createSentTxWorker<T extends EstimationType>(gasPrice: Gas
         // Validation of these jobs will be done in `poolTxWorker`
         const waitingJobIds = []
         const reschedulePromises = []
+        const newPoolJobIdMapping: Record<string, string> = {}
         const waitingJobs = await sentTxQueue.getJobs(['delayed', 'waiting'])
         for (let wj of waitingJobs) {
           // One of the jobs can be undefined, so we need to check it
           // https://github.com/taskforcesh/bullmq/blob/master/src/commands/addJob-8.lua#L142-L143
           if (!wj?.id) continue
           waitingJobIds.push(wj.id)
-          reschedulePromises.push(poolTxQueue.add(txHash, [wj.data.txPayload]).then(j => j.id as string))
+          const reschedulePromise = poolTxQueue.add(txHash, [wj.data.txPayload]).then(j => {
```

Comment:
What will happen if one of Redis requests will fail, but others won't?

I guess it would be better if we use batch/atomic transactions to make group actions in places like this. I assume there are many similar places in the code, so let's do it under a separate issue/PR later on.

Reply:
In this case we firstly add new jobs to the queue, then update `jobIdsMapping`. So, job re-scheduling do not depend on id mapping update. Do you suggest to make these two actions atomic? Also, how would we handle such redis requests errors?

Reply:
It's a good practice to make all db updates during worker iteration within atomic transaction. Case when we re-schedule the job, but don't update the id mapping is not good.

Any error within the worker iteration, which cannot be handled, should rollback the current transaction, as if nothing has ever happened.

Reply:
I will check if it is possible to easily integrate atomic transactions with `bullmq`.

Reply:
Thanks, it is not a big problem anyway, no need to address it in this PR